src/Server/passport/lib/app/lifecycle.ts

```diff
@@ -31,7 +31,35 @@ module.exports.onLocalLogin = function (
   passport.authenticate("local-with-otp")(req, res, function (err: any) {
     if (req.xhr) {
       if (err) {
-        return res.status(500).send({ success: false, error: err.message })
+        switch (true) {
+          case err.message?.includes("invalid email or password"): {
+            return res.status(403).send({ success: false, error: err.message })
```

Comment:
suggestion: After reviewing the [status codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses), I believe sending a `401` here would be more appropriate seeing as the client is (likely) unauthenticated at this point.

Same applies to L41 and L46 when responding to _missing_ or _invalid_ 2fa code.

Reply:
Ah nice, updated 👍 

Also gonna add some quick test coverage.